pages/instructorAssessments/instructorAssessments.js

```diff
@@ -35,40 +37,51 @@ const fileSubmissionsName = (locals) => {
 
 const fileSubmissionsFilename = (locals) => `${fileSubmissionsName(locals)}.zip`;
 
-router.get('/', function (req, res, next) {
-  res.locals.csvFilename = csvFilename(res.locals);
-  res.locals.fileSubmissionsFilename = fileSubmissionsFilename(res.locals);
-  var params = {
-    course_instance_id: res.locals.course_instance.id,
-    authz_data: res.locals.authz_data,
-    req_date: res.locals.req_date,
-    assessments_group_by: res.locals.course_instance.assessments_group_by,
-  };
-  sqldb.query(sql.select_assessments, params, function (err, result) {
-    if (ERR(err, next)) return;
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    res.locals.csvFilename = csvFilename(res.locals);
+    res.locals.fileSubmissionsFilename = fileSubmissionsFilename(res.locals);
+
+    // update assessment statistics if needed
+    assessment.updateAssessmentStatistics(res.locals.course_instance.id);
+
+    var params = {
+      course_instance_id: res.locals.course_instance.id,
+      authz_data: res.locals.authz_data,
+      req_date: res.locals.req_date,
+      assessments_group_by: res.locals.course_instance.assessments_group_by,
+    };
+    const result = sqldb.queryAsync(sql.select_assessments, params);
 
     res.locals.rows = _.map(result.rows, (row) => {
       if (row.sync_errors) row.sync_errors_ansified = ansiUp.ansi_to_html(row.sync_errors);
       if (row.sync_warnings) row.sync_warnings_ansified = ansiUp.ansi_to_html(row.sync_warnings);
       return row;
     });
+
     res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-  });
-});
+  })
+);
 
-router.get('/:filename', function (req, res, next) {
-  if (req.params.filename === csvFilename(res.locals)) {
-    // There is no need to check if the user has permission to view student
-    // data, because this file only has aggregate data.
+router.get(
+  '/:filename',
+  asyncHandler(async (req, res) => {
+    if (req.params.filename === csvFilename(res.locals)) {
+      // There is no need to check if the user has permission to view student
+      // data, because this file only has aggregate data.
+
+      // update assessment statistics if needed
+      assessment.updateAssessmentStatistics(res.locals.course_instance.id);
+
+      var params = {
+        course_instance_id: res.locals.course_instance.id,
+        authz_data: res.locals.authz_data,
+        req_date: res.locals.req_date,
+        assessments_group_by: res.locals.course_instance.assessments_group_by,
+      };
+      const result = sqldb.queryAsync(sql.select_assessments, params);
```

Comment:
Shouldn't it be await? 🙂

Reply:
Hahaha, this has to the most inefficient method of debugging code ever 😄

Fixed in 3dc5fa7396b0585d69f6809972b8e800ea44a4f8 (but still not actually tested 😅)